backend/src/routes/v2/secrets.ts

```diff
@@ -0,0 +1,177 @@
+                        !secret.secretKeyIV ||
+                        !secret.secretKeyTag ||
+                        !secret.secretValueCiphertext ||
+                        !secret.secretValueIV ||
+                        !secret.secretValueTag
+                    ) {
+                        throw new Error('secrets array must contain objects that conform to the Secret interface');
+                    }
+                }
+            } else if (typeof value === 'object') {
+                // case: update 1 secret
+                if (
+                    !value.type || 
+                    !(value.type === SECRET_PERSONAL || value.type === SECRET_SHARED) ||
+                    !value.secretKeyCiphertext ||
+                    !value.secretKeyIV ||
+                    !value.secretKeyTag ||
+                    !value.secretValueCiphertext ||
+                    !value.secretValueIV ||
+                    !value.secretValueTag
+                ) {
+                    throw new Error('secrets array must contain objects that conform to the Secret interface');
+                } 
+            } else {
+                throw new Error('secrets must be an object or an array of objects')
+            }
+            
+            return true;
+    }),
+    validateRequest,
+    requireAuth({
+        acceptedAuthModes: ['jwt']
+    }),
+    requireWorkspaceAuth({
+        acceptedRoles: [ADMIN, MEMBER],
+        location: 'body'
+    }),
+    secretsController.createSecrets
+);
+
+router.get(
+    '/',
+    query('workspaceId').exists().trim(),
+    query('environment').exists().trim().isIn(['dev', 'staging', 'prod', 'test']),
+    validateRequest,
+    requireAuth({
+        acceptedAuthModes: ['jwt', 'serviceToken']
+    }),
+    requireWorkspaceAuth({
+        acceptedRoles: [ADMIN, MEMBER],
+        location: 'query'
+    }),
+    secretsController.getSecrets
+);
+
+router.patch(
+    '/',
+    body('workspaceId').exists().trim(),
+    body('environment').exists().trim().isIn(['dev', 'staging', 'prod', 'test']),
+    body('secrets')
+        .exists()
+        .custom((value) => {
+            if (Array.isArray(value)) {
+                // case: update multiple secrets
+                if (value.length === 0) throw new Error('secrets cannot be an empty array')
+                for (const secret of value) {
+                    if (
+                        !secret.id || 
+                        !secret.secretKeyCiphertext ||
+                        !secret.secretKeyIV ||
+                        !secret.secretKeyTag ||
+                        !secret.secretValueCiphertext ||
+                        !secret.secretValueIV ||
+                        !secret.secretValueTag
+                    ) {
+                        throw new Error('secrets array must contain objects that conform to the Secret interface');
+                    }
+                }
+            } else if (typeof value === 'object') {
```

Comment:
I have seen this check **a lot** in this PR. If we're gonna use this check over and over again we need to make it seperate function and use that function here. It's really not easy to manage in this scale.

Reply:
Sounds good. I'll throw this check into a helper function 👌